SteamcLog/Classes/Helper/Config.swift

```diff
@@ -21,6 +21,11 @@ public struct Config {
     /// Require that all logged objects conform to Redacted or are all redacted by default.
     @usableFromInline internal let requireRedacted: Bool
 
+    /// Predicate for testing any Error-conforming objects passed in as the associated object at the "error" log level. If true, that specific instance of the error will be downgraded to a warning.
+    /// Useful for stoping certain types of failures (network errors, etc) from being logged off device.
+    let suppressError: ((Error) -> Bool)?
```

Comment:
Yeah, I'd change it to be optional because I was originally trying to set it up so that I could check if it has been set (by checking for nil) inside the library that's using this and set a default set of errors to suppress if not, but that doesn't work the way everything else is structured (you can't change any of the settings after init, and because the logger is a value type, allowing that would cause some strange problems, because the logger is only "shared" between the library and the main app as long as you don't change anything). So I ended up just accepting that the app HAS to install the suppression predicate.

It think the way things are now that probably does make more sense to go back to the old approach of being non-optional and defaulting to `{ _ in false}`